src/js/lightcase.js

```diff
@@ -528,13 +528,20 @@
 							dataType: _self.objectData.requestDataType,
 							data: _self.objectData.requestData,
 							success: function (data, textStatus, jqXHR) {
-								// Unserialize if data is transferred as json
-								if (_self.objectData.requestDataType === 'json') {
-									_self.objectData.data = data;
-								} else {
-									$object.html(data);
+								// Check for X-Ajax-Location
+								if (typeof jqXHR.getResponseHeader('X-Ajax-Location') !== 'undefined' && jqXHR.getResponseHeader('X-Ajax-Location') !== null) {
```

Comment:
Your check is actually mostly the same than the following so it can be simplified to:
```if (jqXHR.getResponseHeader('X-Ajax-Location')) {```